### PR TITLE
Split VCS download() into helper functions for Mercurial

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: downloader/src/funTest/kotlin/MercurialDownloadTest.kt [hunk 1]
```diff
@@ -110,8 +112,10 @@ class MercurialDownloadTest : StringSpec() {
 
         "Mercurial can download based on a version".config(tags = setOf(ExpensiveTag)) {
             val pkg = Package.EMPTY.copy(
-                id = Identifier("Test:::$REPO_VERSION"),
-                vcsProcessed = VcsInfo(VcsType.MERCURIAL, REPO_URL, "")
+                id = Identifier("Test:::$PKG_VERSION"),
+
+                // Use a non-blank dummy revsion to enforce multiple revision candidates being tried.
```
Comment (mnonnenmacher) [inline downloader/src/funTest/kotlin/MercurialDownloadTest.kt]: type: revision

File: downloader/src/funTest/kotlin/MercurialDownloadTest.kt [hunk 2]
```diff
@@ -123,8 +127,10 @@ class MercurialDownloadTest : StringSpec() {
         "Mercurial can download only a single path based on a version"
             .config(enabled = hg.isAtLeastVersion("4.3"), tags = setOf(ExpensiveTag)) {
                 val pkg = Package.EMPTY.copy(
-                    id = Identifier("Test:::$REPO_VERSION"),
-                    vcsProcessed = VcsInfo(VcsType.MERCURIAL, REPO_URL, "", path = REPO_PATH_FOR_VERSION)
+                    id = Identifier("Test:::$PKG_VERSION"),
+
+                    // Use a non-blank dummy revsion to enforce multiple revision candidates being tried.
```
Comment (mnonnenmacher) [inline downloader/src/funTest/kotlin/MercurialDownloadTest.kt]: type: revision

File: downloader/src/funTest/kotlin/MercurialDownloadTest.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/funTest/kotlin/MercurialDownloadTest.kt]: Done.

File: downloader/src/funTest/kotlin/MercurialDownloadTest.kt (hunk 2 above)
Comment (sschuberth) [reply downloader/src/funTest/kotlin/MercurialDownloadTest.kt]: Done.

**Pull request merged**